### PR TITLE
Redo infrastructure and deployment code

### PR DESCRIPTION
**Pull request by hannes-ucsc**
I've tested find-golden-ticket.py against the `dev` deployment created by this branch.

### CODE REVIEW

File: common.mk [hunk 1]
```diff
@@ -0,0 +1,24 @@
+SHELL=/bin/bash
+
+ifndef AZUL_HOME
+$(error Please run "source environment" in the data-store repo root directory before running make commands)
```
Comment (caaespin) [inline common.mk]: data-store. Should be indexer,etc

File: common.mk [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+endif
+
+ifneq ($(shell python -c "import sys; print(sys.version_info >= (3,6))"),True)
+$(error Looks like Python 3.6 is not installed)
+endif
+
+ifneq ($(shell python -c "import sys; print(hasattr(sys, 'real_prefix'))"),True)
```
Comment (caaespin) [inline common.mk]: Swap with the one above

File: common.mk [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+$(error Looks like no virtualenv is active)
+endif
+
+ifneq ($(shell python -c "exec('try: import chalice\nexcept: print(False)\nelse: print(True)')"),True)
+$(error Looks like some or all requirements is missing. Please run 'pip install -r requirements.txt')
```
Comment (caaespin) [inline common.mk]: Say .dev

File: common.mk [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+endif
+
+ifeq ($(shell which terraform),)
+$(warning Looks like TerraForm is not installed. `make deploy` should still work but `make terraform` will not.)
```
Comment (caaespin) [inline common.mk]: Make more elaborate

File: deployments/dev/environment [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Set variables for the `dev` (short for development) deployment here.
+#
+# To define an AZUL_… variable use `_set AZUL_FOO bar`. For all other variables use `export BAR=baz`.
+
+_set AZUL_DEPLOYMENT_STAGE dev
+
+# FIXME: this should probably be pointing at integration
+
+_set AZUL_DSS_ENDPOINT "https://dss.staging.data.humancellatlas.org/v1"
```
Comment (caaespin) [inline deployments/dev/environment]: Make prod as the default.  For all deployments until we are integrated with dcp integration

**Review by caaespin — CHANGES_REQUESTED** [review]
After changing environment variable `make deploy` didn't update the lambda.

File: common.mk (hunk 1 above)
Comment (hannes-ucsc) [reply common.mk]: Fixed.

File: common.mk (hunk 2 above)
Comment (hannes-ucsc) [reply common.mk]: Fixed.

File: common.mk (hunk 3 above)
Comment (hannes-ucsc) [reply common.mk]: Fixed.

File: common.mk (hunk 4 above)
Comment (hannes-ucsc) [reply common.mk]: Fixed.

File: deployments/dev/environment (hunk 5 above)
Comment (hannes-ucsc) [reply deployments/dev/environment]: All DSS endpoint URLs now point at prod. Will commit that separately.

**Comment by hannes-ucsc** [comment]
I addressed all issues.

Ready for re-review, @caaespin.

**Review by caaespin — APPROVED** [review]
LGTM

sealofapproval

**Comment by hannes-ucsc** [comment]
I pushed two two last minute fixes.

**Review by caaespin — APPROVED** [review]
LGTM

**Pull request merged**